domain/modelagent/agentstream.go

```diff
@@ -0,0 +1,57 @@
+// Copyright 2025 Canonical Ltd.
```

Comment:
Not sure if it's worth putting these in the top-level domain package.
It's only used for the model service, might be better to move into the domain model package.

Reply:
Follow up PR's to this will start using it from this package. They will start appearing after this had landed.